lib/system/linux/device.c

```diff
@@ -368,6 +368,16 @@ static struct linux_bus linux_bus[] = {
 				.dev_dma_map = metal_uio_dev_dma_map,
 				.dev_dma_unmap = metal_uio_dev_dma_unmap,
 			},
+			{
+				.drv_name  = "uio_dmem_genirq",
+				.mod_name  = "uio_dmem_genirq",
+				.cls_name  = "uio",
+				.dev_open  = metal_uio_dev_open,
+				.dev_close = metal_uio_dev_close,
+				.dev_irq_ack  = metal_uio_dev_irq_ack,
+				.dev_dma_map = metal_uio_dev_dma_map,
+				.dev_dma_unmap = metal_uio_dev_dma_unmap,
+			},
```

Comment:
From Linux kernel, both uio_dmem and uio_pdrv generic irq are able to enabled together.
the libmetal metal_linux_dev_open() it will check the drivers in the drivers list. it will try to open the device with that driver, if it finds the device is already bound with other driver, it will output error but it will continue to try the next one. 

How about:
* opt 1: check the binded drivers before blindly check through the drivers defined
* opt 2: change the metal_uio_dev_bind() to output INFO instead of ERROR when the device is binded with other driver and return -EBUSY.

Opt2 is simpler.

Reply:
Add a change for option 2